tests/components/wilight/test_fan.py

```diff
@@ -0,0 +1,205 @@
+"""Tests for the WiLight integration."""
+from asynctest import patch
+import pytest
+import pywilight
+
+from homeassistant.components.fan import (
+    ATTR_DIRECTION,
+    ATTR_SPEED,
+    DIRECTION_FORWARD,
+    DIRECTION_REVERSE,
+    DOMAIN as FAN_DOMAIN,
+    SERVICE_SET_DIRECTION,
+    SERVICE_SET_SPEED,
+    SPEED_HIGH,
+    SPEED_LOW,
+    SPEED_MEDIUM,
+)
+from homeassistant.const import (
+    ATTR_ENTITY_ID,
+    SERVICE_TURN_OFF,
+    SERVICE_TURN_ON,
+    STATE_OFF,
+    STATE_ON,
+)
+from homeassistant.helpers.typing import HomeAssistantType
+
+from tests.components.wilight import (
+    HOST,
+    UPNP_MAC_ADDRESS,
+    UPNP_MODEL_NAME_LIGHT_FAN,
+    UPNP_MODEL_NUMBER,
+    UPNP_SERIAL,
+    WILIGHT_ID,
+    setup_integration,
+)
+
+
+@pytest.fixture(name="dummy_device_from_host_light_fan")
+def mock_dummy_device_from_host_light_fan():
+    """Mock a valid api_devce."""
+
+    device = pywilight.wilight_from_discovery(
+        f"http://{HOST}:45995/wilight.xml",
+        UPNP_MAC_ADDRESS,
+        UPNP_MODEL_NAME_LIGHT_FAN,
+        UPNP_SERIAL,
+        UPNP_MODEL_NUMBER,
+    )
+
+    device.set_dummy(True)
+
+    with patch(
+        "pywilight.device_from_host",
+        return_value=device,
+    ):
+        yield device
+
+
+async def test_loading_light_fan(
+    hass: HomeAssistantType,
+    dummy_device_from_host_light_fan,
+) -> None:
+    """Test the WiLight configuration entry loading."""
+
+    entry = await setup_integration(hass)
+    assert entry
+    assert entry.unique_id == WILIGHT_ID
+
+    entity_registry = await hass.helpers.entity_registry.async_get_registry()
+
+    # First segment of the strip
+    state = hass.states.get("light.wl000000000099_1")
```

Comment:
We don't need to test the light entity state in the fan tests.

Reply:
Done!